x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,531 @@
+}
+
+interface Environment {
+  name: string;
+  id: string;
+  region: string;
+  es_url: string;
+  kb_url: string;
+  product: string;
+}
+
+interface Credentials {
+  username: string;
+  password: string;
+}
+
+const DEFAULT_REGION = 'aws-eu-west-1';
+let log: ToolingLog;
+
+/**
+ * Retrieve test files using a glob pattern.
+ * If process.env.RUN_ALL_TESTS is true, returns all matching files, otherwise, return files that should be run by this job based on process.env.BUILDKITE_PARALLEL_JOB_COUNT and process.env.BUILDKITE_PARALLEL_JOB
+ */
+const retrieveIntegrations = (integrationsPaths: string[]) => {
+  const nonSkippedSpecs = integrationsPaths.filter((filePath) => !isSkipped(filePath));
+
+  if (process.env.RUN_ALL_TESTS === 'true') {
+    return nonSkippedSpecs;
+  } else {
+    // The number of instances of this job were created
+    const chunksTotal: number = process.env.BUILDKITE_PARALLEL_JOB_COUNT
+      ? parseInt(process.env.BUILDKITE_PARALLEL_JOB_COUNT, 10)
+      : 1;
+    // An index which uniquely identifies this instance of the job
+    const chunkIndex: number = process.env.BUILDKITE_PARALLEL_JOB
+      ? parseInt(process.env.BUILDKITE_PARALLEL_JOB, 10)
+      : 0;
+
+    const nonSkippedSpecsForChunk: string[] = [];
+
+    for (let i = chunkIndex; i < nonSkippedSpecs.length; i += chunksTotal) {
+      nonSkippedSpecsForChunk.push(nonSkippedSpecs[i]);
+    }
+
+    return nonSkippedSpecsForChunk;
+  }
+};
+
+const encode = (str: string): string => Buffer.from(str, 'binary').toString('base64');
+
+const getApiKeyFromElasticCloudJsonFile = () => {
+  const userHomeDir = os.homedir();
+  try {
+    const jsonString = fs.readFileSync(path.join(userHomeDir, '.elastic/cloud.json'), 'utf-8');
+    const jsonData = JSON.parse(jsonString);
+    return jsonData.api_key.qa;
+  } catch (e) {
+    log.info('API KEY could not be found in ');
+    return null;
+  }
+};
+
+// Poller function that is polling every 20s, forever until function is resolved.
+async function poll<T>(
+  fn: () => Promise<T>,
+  retries: number = 200,
+  interval: number = 20000
+): Promise<T> {
+  return Promise.resolve()
+    .then(fn)
+    .catch(async function retry(err: Error): Promise<T> {
+      let remainingRetries = retries;
+      if (remainingRetries-- > 0) {
+        remainingRetries -= 1;
+        return new Promise((resolve) => setTimeout(resolve, interval)).then(fn).catch(retry);
+      }
+      throw err;
+    });
+}
+
+// Method to invoke the create environment API for serverless.
+async function createEnvironment(
+  baseUrl: string,
+  projectName: string,
+  runnerId: string,
+  apiKey: string,
+  ftrConfig: SecuritySolutionDescribeBlockFtrConfig,
+  onEarlyExit: (msg: string) => void
+): Promise<Environment> {
+  log.info(`${runnerId}: Creating environment ${projectName}...`);
+  const environment = {} as Environment;
+  const body = {
+    name: projectName,
+    region_id: DEFAULT_REGION,
+  } as CreateEnvironmentRequestBody;
```

Comment:
`product_types` field is a non optional one. Does an empty array work here? If not the type `CreateEnvironmentRequestBody` should be changed to define `product_types` as optional.

```suggestion
    product_types: ftrConfig?.productTypes ?? [],
  };
```

Reply:
Fixed in commit : https://github.com/elastic/kibana/pull/169311/commits/d7613c83c18377ed29ec117643d912a047215e32